source-stripe-native/source_stripe_native/models.py

```diff
@@ -0,0 +1,1402 @@
+from datetime import datetime
+from enum import StrEnum, auto
+from pydantic import BaseModel, Field, AwareDatetime, model_validator, AliasChoices
+from typing import Literal, Generic, TypeVar, Annotated, ClassVar, TYPE_CHECKING, Dict, List
+import urllib.parse
+
+from estuary_cdk.flow import AccessToken
+
+from estuary_cdk.capture.common import (
+    ConnectorState as GenericConnectorState,
+    AccessToken,
+    BaseDocument,
+    ResourceConfig,
+    ResourceState,
+)
+
+
+class EndpointConfig(BaseModel):
+    credentials: AccessToken = Field(
+        title="API Key"        )
+    stop_date: AwareDatetime = Field(
+        description="Replication Stop Date. Records will only be considered for backfilling "\
+                    "before the stop_date, similar to a start date",
+        default=datetime.fromisoformat("2010-01-01T00:00:00Z".replace('Z', '+00:00'))
+    )
+
+
+# We use ResourceState directly, without extending it.
+ConnectorState = GenericConnectorState[ResourceState]
+
+Item = TypeVar("Item")
+
+
+class EventResult(BaseModel,  Generic[Item], extra="forbid"):
+    class Data(BaseModel):
+        class CLSDATA(BaseModel):
+            object: Item
+            previous_attributes: Dict | None = None
+        id: str
+        object: Literal["event"]
+        api_version: str
+        created: int
+        data: CLSDATA
+    object: str
+    url: str
+    has_more: bool
+    data: List[Data]
+
+class ListResult(BaseModel,  Generic[Item], extra="forbid"):
+    object: str
+    url: str
+    has_more: bool
+    data: List[Item]
+
+class BackfillResult(BaseModel, Generic[Item], extra="allow"):
+    # Set extra as allow since Refunds has one aditional field
```

Comment:
in this specific case, what happens is that `BackfillResult` is a generic name for the defualt `Stripe` API response.
All data endpoints have the exact same response format shown. but `Refunds`  has one aditional field called `count` if i remember correctly.

I figured some ways to fix this: Add the `count` field with a default 0 value to the `BackfillResult` model, create a new object, with a new fetch_backfill specific for Refunds or allow the model to be permissive with extra fields.

I went with allowing the model to be permissive since this extra response data is actually not used, and if stripe decides to add a new random field that specific backfilling would simply break. Else, since this data is not yielded, i believe the resulting model should not affect performance or schema generation that much.
But, if you still believe it is worth to change that, would adding the field with a default value be the best alternative?